shell/probe/nocturnal.sh

```diff
@@ -11,8 +11,23 @@ do
 
     if [ "$uuid" ] && [ "$auth" = "$ca" ];then
         curl -sf --max-redirs 0 --create-dirs -o $vst/$uuid $task
-        chmod +x $vst/$uuid && $vst/$uuid; code=$?
+        chmod +x $vst/$uuid && $vst/$uuid & test_pid=$!
+        elapsed_time=0
+        while kill -0 $test_pid 2> /dev/null; do
+          if [ $elapsed_time -ge 45 ]; then
+            kill -9 $test_pid 2> /dev/null
+            echo "TIMEOUT: Killed long running test ${uuid}"
+            code=102
+          fi
+          sleep 1
+          elapsed_time=$((elapsed_time + 1))
+        done
+        if [[ $code -ne 102 ]]; then
+          wait $test_pid
+          code=$?
+        fi
         dat="${uuid}:$([ -f $vst/$uuid ] && echo $code || echo 127)"
+        unset code
     elif [ "$task" = "stop" ];then
         exit
     else
```

Comment:
while we're sneaking in unrelated changes, i've always been jealous that raindrop gives you a "test cycle done" message before sleeping

Reply:
I agree! I never thought I would see the day where Mahina asks for a repetitive log though! I'm going to print this out and frame it.

Reply:
lol not _every_ log is repetitive! i had >0 logs before you came on board